Tasks/DotNetCoreInstallerV0/dotnetcoreinstaller.ts

```diff
@@ -3,28 +3,80 @@ import * as toolLib from 'vsts-task-tool-lib/tool';
 import * as trm from 'vsts-task-lib/toolrunner';
 import { DotNetCoreReleaseFetcher } from "./releasesfetcher";
 import * as utilities from "./utilities";
-
+import * as fileSystem from "fs";
 import * as os from 'os';
 import * as path from 'path';
 
 class DotnetCoreInstaller {
-    constructor(packageType, version) {
+    constructor(workingDirectory: string, packageType: string, version?: string, useGlobalJson?: boolean) {
+        this.useGlobalJson = useGlobalJson;
         this.packageType = packageType;
+        this.workingDirectory = workingDirectory;
         if (!toolLib.isExplicitVersion(version)) {
             throw tl.loc("ImplicitVersionNotSupported", version);
         }
+        if(!useGlobalJson && (version == null || version.length == 0)){
+            throw tl.loc("VersionMissingException", "If you don't use global.json then please defined a version.")
+        }
         this.version = version;
         this.cachedToolName = this.packageType === 'runtime' ? 'dncr' : 'dncs';;
     }
 
-    public async install() {
-        // Check cache
-        let toolPath: string;
-        let osSuffixes = this.detectMachineOS();
-        let parts = osSuffixes[0].split("-");
-        this.arch = parts.length > 1 ? parts[1] : "x64";
-        toolPath = this.getLocalTool();
+    private async installFromGlobalJson(osSuffixes: string[]) {
+        let filePathsToGlobalJson = this.getFiles(this.workingDirectory ? this.workingDirectory : ".", "global.json");
+        if(filePathsToGlobalJson.length == 0){
+            throw tl.loc("FailedToFindGlobalJson", "From the root of your working folder we can't find any global.json file.");
+        }
+        let sdkVersionNumber = new Array<{ name: string, toolPath?: string }>();
```

Comment:
This might be better =>
let sdkVersions = new Array<{ version: string, toolPath?: string }>();

Reply:
done.

Reply:
I think we have resolve this. If not reopen this conversion.